File: leetcode/334.py
import sys


class Solution:
    def increasingTriplet(self, nums: list[int]) -> bool:
        n1, n2 = sys.maxsize, sys.maxsize
        for n in nums:
            if n > n2:
                return True
            if n <= n1:
                n1 = n
            elif n <= n2:
                n2 = n
        return False

print(Solution.increasingTriplet([5,4,3,2,1]))

File: leetcode/334.dart
class Solution {
  bool increasingTriplet(List<int> nums) {
    int n1 = 100000000;
    int n2 = 100000000;
    for (int n in nums) {
      if (n > n2) {
        return true;
      }
      if (n <= n1) {
        n1 = n;
      } else if (n <= n2) {
        n2 = n;
      }
    }
    return false;
  }
}
